nni/algorithms/compression/pytorch/pruning/admm_pruner.py

```diff
@@ -106,31 +106,14 @@ def _projection(self, weight, sparsity):
             original matrix
         sparsity : float
             the ratio of parameters which need to be set to zero
-
+        wrapper: PrunerModuleWrapper
+            layer wrapper of this layer
         Returns
         -------
         tensor
             the projected matrix
         '''
-        w_abs = weight.abs()
-        if self._base_algo == 'level':
-            k = int(weight.numel() * sparsity)
-            if k == 0:
-                mask_weight = torch.ones(weight.shape).type_as(weight)
-            else:
-                threshold = torch.topk(w_abs.view(-1), k, largest=False)[0].max()
-                mask_weight = torch.gt(w_abs, threshold).type_as(weight)
-        elif self._base_algo in ['l1', 'l2']:
-            filters = weight.size(0)
-            num_prune = int(filters * sparsity)
-            if filters < 2 or num_prune < 1:
-                mask_weight = torch.ones(weight.size()).type_as(weight).detach()
-            else:
-                w_abs_structured = w_abs.view(filters, -1).sum(dim=1)
-                threshold = torch.topk(w_abs_structured.view(-1), num_prune, largest=False)[0].max()
-                mask_weight = torch.gt(w_abs_structured, threshold)[:, None, None, None].expand_as(weight).type_as(weight)
-
-        return weight.data.mul(mask_weight)
+        return weight.data.mul(self.masker.calc_mask(sparsity, wrapper)['weight_mask'])
```

Comment:
It seems that `self.masker.calc_mask(sparsity, wrapper)` will calculate mask based on `wrapper.module.weight` instead of the variable `weight`. Could you please check that ?

Reply:
good catch,

Reply:
Fixed